internal/mapper/datasource_mapper.go

```diff
@@ -72,9 +73,24 @@ func generateDataSourceSchema(logger *slog.Logger, dataSource explorer.DataSourc
 	if err != nil {
 		return nil, err
 	}
-	readResponseAttributes, schemaErr := readResponseSchema.BuildDataSourceAttributes()
-	if schemaErr != nil {
-		return nil, schemaErr
+
+	readResponseAttributes := attrmapper.DataSourceAttributes{}
+	if readResponseSchema.Type == util.OAS_type_array {
+		logger.Debug(fmt.Sprintf("response body is an array, building '%s' collection attribute", name))
+
+		collectionAttribute, schemaErr := readResponseSchema.BuildDataSourceAttribute(name, schema.Computed)
```

Comment:
Looking for some thoughts on the default name for a collection data source. I chose the name of the data source.

Other considered options:
- Static string: `results`
- Static string: `responses`

Reply:
Using the name of the data source seems fine to me.

Reply:
Agree with Ben, some providers already use this design, such as cloudflare/cloudflare.

One thing to maybe create an issue  for in the future is that data sources might include multiple "words" in the name, e.g. [hashicorp/aws provider aws_secretsmanager_secrets](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/secretsmanager_secrets) -- in those cases I'm guessing it should split on underscores and use only the last "word".

Reply:
At that point, maybe should just allow configuration of the name 😆 . I will create an issue to document the future issue 👍🏻 

```yaml
data_sources:
  petmanager_pets:
    read:
      path: /pet/findByStatus
      method: GET
    schema:
      collection_name: pets 
```

Reply:
Created #83 to consider this issue later!